File: 模板/python/线性筛.py
def euler(limit: int) -> list[int]:
    limit += 1
    isPrime = [True] * limit
    primeList = []
    for i in range(2, limit):
        if isPrime[i]:
            primeList.append(i)
        for prime in primeList:
            temp = i * prime
            if temp >= limit:
                break
            isPrime[temp] = False
            if i % prime == 0:
                break
    return primeList

File: 模板/数学/线性筛.cpp
#include <vector>

using namespace std;

vector<int> euler(int limit) {
    ++limit;
    vector<bool> isPrime(limit, true);
    vector<int> primeList;
    for (int i = 2; i < limit; ++i) {
        if (isPrime[i]) {
            primeList.push_back(i);
        }
        for (int p : primeList) {
            if ((long long) i * p >= limit) {
                break;
            }
            isPrime[i * p] = false;
            if (i % p == 0) {
                break;
            }
        }
    }
    return primeList;
}